frontend/src/libhoclient/host_orchestrator_client.go

```diff
@@ -465,10 +468,16 @@ func asWebRTCICEServers(in []opapi.IceServer) []webrtc.ICEServer {
 }
 
 func addBuildAPICredsHeaders(rb *HTTPRequestBuilder, creds BuildAPICredential) {
-	if creds.AccessToken != "" {
-		rb.AddHeader(DefaultHostOrchestratorCredentialsHeader, creds.AccessToken)
-		if creds.UserProjectID != "" {
-			rb.AddHeader(BuildAPICredsUserProjectIDHeader, creds.UserProjectID)
-		}
+	if creds.AccessToken == "" {
+		return
+	}
+	if creds.AccessToken == "inject" {
```

Comment:
nit: Instead of having 3 different behaviors based on the value it's better to have two functions:
```
func AddBuildCredsHeaders(rb *HTTPRequestBuilder, creds BuildAPICredential) {
  // Add the access token and project id (if not empty)
}

func requestBuildAPICredsInjection(rb *HTTPRequestBuilder) {
    // Adds the inject header
}
```

The caller probably knows which behavior it wants as it needs to handle refreshing the tokens when not injecting, so this approach is less error prone.

Reply:
agree that having two helpers is less error prone, however I had to add a new helper `handleBuildAPICredsHeaders` to avoid repeating the same code in the callers.

Reply:
I see, that's still better though, as the function has a clear purpose easy to understand from the code.